### ISSUE TITLE
Double encoding for SKUs in GET queries

### ISSUE THREAD

**Issue by YuriiHerasymchuk** [amz-tools/amazon-sp-api#134]
Amazon expects double URL encoding for GET requests with a list of SKUs in the query (i.e. `/products/pricing/v0/price`).
The library encodes it only once and because of that when you have an SKU with special characters Amazon returns an error.

**Example: sku with special character & - 'Test&Sku'**

When encoding it once - Amazon does not recognize the SKU:
[https://sellingpartnerapi-na.amazon.com/products/pricing/v0/price?ItemType=Sku&MarketplaceId=ATVPDKIKX0DER&Skus=Test%26Sku](https://sellingpartnerapi-na.amazon.com/products/pricing/v0/price?ItemType=Sku&MarketplaceId=ATVPDKIKX0DER&Skus=Test%26Sku)
response: {"body":"\n{\n\"payload\": [\n    {\n  \"status\": \"ClientError\"\n        ,\"SellerSKU\": \"Test\"\n              }\n      ]\n}\n","statusCode":200}

With double-encoded SKU - successful response with information:
[https://sellingpartnerapi-na.amazon.com/products/pricing/v0/price?ItemType=Sku&MarketplaceId=ATVPDKIKX0DER&Skus=Test%2526Sku](https://sellingpartnerapi-na.amazon.com/products/pricing/v0/price?ItemType=Sku&MarketplaceId=ATVPDKIKX0DER&Skus=Test%2526Sku)

Let me know if you have encountered the same problem. I can create a PR to fix this issue.

**Comment by amz-tools** [comment]
Hi @YuriiHerasymchuk,

we do have the same issue when trying the call. It doesn't work single encoded but works fine when double encoded.

However I am very sure we fixed the issue in the past and I think we discovered a bigger problem here. The handling seems to be inconsistent when comparing it with other operations. 

I.e. the `listCatalogItems` operation works fine when the `SellerSKU` is single encoded, but doesn't work when double encoded. When trying the below call with a valid SKU that includes a `&` it will return an items array including the item information only when single encoded but an empty array when double encoded.

```
await sellingPartner.callAPI({
  operation:'catalogItems.listCatalogItems',
  query:{
    MarketplaceId:'A1805IZSGTT6HS',
    SellerSKU:'A VALID TEST SKU WITH &'
  }
});
```
PLEASE NOTE: Trying with a test sku will always return an empty items array so in order to test it you would require an actual SKU with a `&`.

To sum up, I am not quite sure what to do here, maybe we should open a Sellingpartner docs issue and/or we would probably have to define for each operation that includes an SKU as query parameter if it needs to be single or double encoded.

**Comment by amz-tools** [comment]
We tested all operations with SKU in query for single/double encoding and it turns out that only the operations of the `productPricing` endpoint expects the value to be double encoded. We fixed this now in v0.7.7.
